src/dso_api/dynamic_api/filterset.py

```diff
@@ -37,29 +49,61 @@ def get_filters(cls):
 
         # Apply camelCase to the filter names, after they've been initialized
         # using the model fields snake_case as input.
-        return {
-            snake_to_camel_case(attr_name): filter
-            for attr_name, filter in filters.items()
+        base_filters = {
+            snake_to_camel_case(attr_name): filter_class
+            for attr_name, filter_class in filters.items()
         }
 
+        return base_filters
 
-def filterset_factory(model: Type[DynamicModel]) -> Type[DynamicFilterSet]:
+
+def filterset_factory(model: Type[DynamicModel], serializer_class: Type[DynamicSerializer]) -> Type[DynamicFilterSet]:
     """Generate the filterset based on the dynamic model."""
     # See https://django-filter.readthedocs.io/en/master/guide/usage.html on how filters are used.
     # Determine which fields are included:
     # Excluding geometry fields for now, as the default filter only performs exact matches.
     # This isn't useful for polygon fields, and excluding it avoids support issues later.
-    fields = {f.attname: _get_field_lookups(f) for f in model._meta.get_fields()}
+    fields = {f.attname: _get_field_lookups(f.__class__) for f in model._meta.get_fields()}
+
+    filters = {}
+    camel_case_fields = list(map(snake_to_camel_case, fields))
+    for field_name in serializer_class.Meta.fields:
+        if field_name not in ["_links", "schema"] + camel_case_fields:
+            try:
+                field_schema = model._table_schema['schema']['properties'][field_name]
+            except KeyError:
+                pass
+            else:
+                if field_is_nested_table(field_schema):
+                    for inner_name, inner_field in field_schema['items']['properties'].items():
+                        model_field = JSON_TYPE_TO_DJANGO[inner_field['type']][0]
+                        filter_class = DSOFilterSet.FILTER_DEFAULTS.get(model_field)
+                        if filter_class is not None:
+                            filter_class = filter_class["filter_class"]
+                            filter_name = "__".join([field_name, format_field_name(inner_name)])
+                            field_lookups = _get_field_lookups(model_field)
+                            for lookup_expr in field_lookups:
+                                subfilter_name = filter_name
+                                if lookup_expr != 'exact':
+                                    subfilter_name = f"{filter_name}__{lookup_expr}"
+                                filters[subfilter_name] = filter_class(
+                                    field_name="__".join([field_name, slugify(inner_name, sign='_')]),
+                                    lookup_expr=lookup_expr if lookup_expr != 'exact' else 'contains',
+                                    label=DSOFilterSet.FILTER_HELP_TEXT.get(filter_class, lookup_expr)
+                                )
+                                fields[subfilter_name] = field_lookups
```

Comment:
Dit hele blok zou ik in een losse functie zetten.

Een paar comments erbij wat hier gebeurt is ook wel prettig :-)

Reply:
Ja, paar comments zijn wel idd handig :)